src/track/beats.h

```diff
@@ -54,124 +84,264 @@ class Beats : public QObject {
         THREEHALVES,
     };
 
-    virtual Beats::CapabilitiesFlags getCapabilities() const = 0;
-
-    // Serialization
-    virtual QByteArray toByteArray() const = 0;
-    virtual BeatsPointer clone() const = 0;
-
-    // A string representing the version of the beat-processing code that
-    // produced this Beats instance. Used by BeatsFactory for associating a
-    // given serialization with the version that produced it.
-    virtual QString getVersion() const = 0;
-    // A sub-version can be used to represent the preferences used to generate
-    // the beats object.
-    virtual QString getSubVersion() const = 0;
-
-    ////////////////////////////////////////////////////////////////////////////
-    // Beat calculations
-    ////////////////////////////////////////////////////////////////////////////
+    // TODO(JVC) Not needed
+    Beats::CapabilitiesFlags getCapabilities() const {
+        return BEATSCAP_TRANSLATE | BEATSCAP_SCALE | BEATSCAP_ADDREMOVE |
+                BEATSCAP_MOVEBEAT | BEATSCAP_HASBAR;
+    }
+    /// Serializes into a protobuff.
+    virtual QByteArray toProtobuff() const;
+    virtual BeatsPointer clone() const;
+
+    /// Returns a string representing the version of the beat-processing code that
+    /// produced this Beats instance. Used by BeatsFactory for associating a
+    /// given serialization with the version that produced it.
+    virtual QString getVersion() const;
+    /// Return a string that represent the preferences used to generate
+    /// the beats object.
+    virtual QString getSubVersion() const;
+    virtual void setSubVersion(QString subVersion);
```

Comment:
I don't understand the purpose of this subversion concept. The version specifies the protobuf version so we can upgrade old data from the database to new versions in the future. What does a subversion add to that? I think we can remove it.

Reply:
I am not sure how database migrations will work if we remove the subversion concept (along with versions as well).
Can this lead to problems?